src/math/p5.Vector.js

```diff
@@ -1349,41 +1417,133 @@ p5.Vector = class {
   }
 
   /**
- * Set the magnitude of this vector to the value used for the `len`
- * parameter.
+ * Sets a vector's magnitude to a given value.
+ *
+ * The static version of `setMag()`, as in `p5.Vector.setMag(v, 10)`, returns
+ * a new <a href="#/p5.Vector">p5.Vector</a> object and doesn't change the
+ * original.
  *
  * @method setMag
- * @param  {number}    len The new length for this vector
+ * @param  {number}    len new length for this vector.
  * @chainable
  * @example
  * <div class="norender">
  * <code>
  * let v = createVector(3, 4, 0);
- * // v has components [3.0, 4.0, 0.0]
+ * // Prints "5" to the console.
+ * print(v.mag());
+ *
  * v.setMag(10);
- * // v's components are set to [6.0, 8.0, 0.0]
+ * // Prints "p5.Vector Object : [6, 8, 0]" to the console.
+ * print(v.toString());
+ * </code>
+ * </div>
+ *
+ * <div class="norender">
+ * <code>
+ * let v0 = createVector(3, 4, 0);
+ * let v1 = p5.Vector.setMag(v0, 10);
+ * // Prints "5" to the console.
+ * print(v0.mag());
+ * // Prints "p5.Vector Object : [6, 8, 0]" to the console.
+ * print(v1.toString());
+ * </code>
+ * </div>
+ *
+ * <div>
+ * <code>
+ * function draw() {
+ *   background(240);
+ *
+ *   let origin = createVector(0, 0);
+ *   let v = createVector(50, 50);
+ *
+ *   drawArrow(origin, v, 'red');
+ *
+ *   v.setMag(30);
+ *   drawArrow(origin, v, 'blue');
+ *
+ *   describe('Two arrows extend from the top left corner of a square toward its center. The red arrow reaches the center and the blue arrow only extends part of the way.');
+ * }
+ *
+ * function drawArrow(base, vec, myColor) {
+ *   push();
+ *   stroke(myColor);
+ *   strokeWeight(3);
+ *   fill(myColor);
+ *   translate(base.x, base.y);
+ *   line(0, 0, vec.x, vec.y);
+ *   rotate(vec.heading());
+ *   let arrowSize = 7;
+ *   translate(vec.mag() - arrowSize, 0);
+ *   triangle(0, arrowSize / 2, 0, -arrowSize / 2, arrowSize, 0);
+ *   pop();
+ * }
+ * </code>
+ * </div>
+ */
+  setMag(n) {
+    return this.normalize().mult(n);
+  }
+
+  /**
+ * Calculates the angle a 2D vector makes with the positive x-axis. Angles
+ * increase in the clockwise direction.
```

Comment:
Oh great call adding this, this is the &num; 1 thing I forget about angles! Do you think it's worth mentioning the clockwise direction in `setHeading` and `rotate` too?

Reply:
Sure do, added a note to both.